### PR TITLE
Ar 755 group selected study series

### PR DESCRIPTION
**Pull request by davgur**
https://docs.google.com/spreadsheets/d/1ss6wwIIksUhCw_5ypTiHRRwSlTwu4bs0kF6JOTgkDrQ/edit#gid=1849713197

Test:
http://bbdev6.kbb1.com:4502/he/search?q=%D7%A4%D7%AA%D7%99%D7%97%D7%94%20%D7%9C%D7%97%D7%9B%D7%9E%D7%AA%20%D7%94%D7%A7%D7%91%D7%9C%D7%94

### CODE REVIEW

File: consts/consts.go [hunk 1]
```diff
@@ -392,6 +392,7 @@ const ES_RESULT_TYPE_TWEETS = "tweets"
 
 // Result of many tweets in one hit
 const SEARCH_RESULT_TWEETS_MANY = "tweets_many"
+const SEARCH__RESULT_LESSONS_SERIES = "lessons_series"
```
Comment (YuriHechter) [inline consts/consts.go:395]: remove one underscore

File: search/query.go [hunk 2]
```diff
@@ -206,6 +206,20 @@ func createSpanNearQuery(field string, term string, boost float32, slop int, inO
 	return query, nil
 }
 
+func addMastNotSeries(q Query) *elastic.BoolQuery {
```
Comment (YuriHechter) [inline search/query.go:209]: addMustNotSeries

File: consts/consts.go (hunk 1 above)
Comment (davgur) [reply consts/consts.go]: fixed

File: search/query.go (hunk 2 above)
Comment (davgur) [reply search/query.go]: fixed

File: search/query.go [hunk 3]
```diff
@@ -206,6 +206,20 @@ func createSpanNearQuery(field string, term string, boost float32, slop int, inO
 	return query, nil
 }
 
+func addMustNotSeries(q Query) *elastic.BoolQuery {
```
Comment (bbfsdev) [inline search/query.go:209]: Please comment this function, what this does and for what reason.

File: search/quality_eval.go [hunk 4]
```diff
@@ -1476,6 +1477,11 @@ func evalResultToHitSources(result EvalResult) ([]HitSource, error) {
 					}
 					hitSource.CarrouselHitSources = append(hitSource.CarrouselHitSources, carrouselHitSource)
 				}
+			} else if hit.Type == consts.SEARCH_RESULT_LESSONS_SERIES_BY_SOURCE || hit.Type == consts.SEARCH_RESULT_LESSONS_SERIES_BY_TAG {
```
Comment (bbfsdev) [inline search/quality_eval.go:1480]: You generate a new result type here?
What about frontend? How it looks like?

File: search/lessons_series_engine.go [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+	"time"
+
+	"github.com/Bnei-Baruch/archive-backend/consts"
+	"github.com/Bnei-Baruch/archive-backend/es"
+	"github.com/Bnei-Baruch/archive-backend/utils"
+	"github.com/pkg/errors"
+	"gopkg.in/olivere/elastic.v6"
+)
+
+func (e *ESEngine) LessonsSeries(query Query, preference string) (map[string]*elastic.SearchResult, error) {
```
Comment (bbfsdev) [inline search/lessons_series_engine.go:17]: Please add function description + example.

**Review by bbfsdev — CHANGES_REQUESTED** [review]

**Comment by bbfsdev** [comment]
Please provide demo / client change for this.

File: cmd/index.go [hunk 6]
```diff
@@ -406,3 +406,9 @@ func simulateUpdateFn(cmd *cobra.Command, args []string) {
 	log.Info("Success")
 	log.Infof("Total run time: %s", time.Now().Sub(clock).String())
 }
+
+func getDateAlias() string {
+	t := time.Now()
+	date := strings.ToLower(t.Format(time.RFC3339))
+	return strings.ReplaceAll(date, "+", "p")
```
[the remark this answers is not in the document]
Comment (davgur) [reply cmd/index.go:413]: > What problem this solves? Is this correct for all OS (windows/linux) as I remember some difference here. This point is delicate.

I dont have this problemm now. Look like it was for specific OS only. What you say? remove it or not?

File: search/engine.go [hunk 7]
```diff
@@ -579,6 +579,7 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
 	grammarsFilterIntentsChannel := make(chan []Intent, 1)
 	grammarsFilteredResultsByLangChannel := make(chan map[string][]FilteredSearchResult)
 	tweetsByLangChannel := make(chan map[string]*elastic.SearchResult)
+	seriesLangChannel := make(chan map[string]*elastic.SearchResult)
```
[the remark this answers is not in the document]
Comment (davgur) [reply search/engine.go:582]: what is mean semantic?

File: search/quality_eval.go (hunk 4 above)
Comment (davgur) [reply search/quality_eval.go:1480]: https://github.com/Bnei-Baruch/kmedia-mdb/tree/AR-755_group_selected_study_series_in_search_results

File: cmd/index.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (bbfsdev) [reply cmd/index.go:413]: Keep it should work well with everyone.

**Review by bbfsdev — APPROVED** [review]

**Review by YuriHechter — APPROVED** [review]

**Pull request merged**